### ISSUE TITLE
update raster source url?

### ISSUE THREAD

**Issue by wendellwt** [eoda-dev/py-maplibregl#111]
How do I update the url in a raster source?

This works fine when the layer source is GeoJSONSource:

`m.set_data(JSON_LAYER_ID, new_data)  # works great, but only for GeoJSON source`

is there an equivalent for raster layers?  Maybe something like:

`m.set_raster_url(RASTER_LAYER_ID, new_url)   # no such method`

I tried this, but it doesn't do anything useful:

`m.set_layout_property(RASTER_LAYER_ID, 'url', new_url)   # nope`

Thank you.

**Comment by crazycapivara** [comment]
You have to remove and re-add the layer with the new url.

**Comment by crazycapivara** [comment]
You are right. There are not mappings for all JavaScript Map methods but with the powerful [Map.add_call](https://eoda-dev.github.io/py-maplibregl/api/map/#maplibre.Map.add_call) method you can trigger any `maplibregl.Map` method. The first argument is the name of the JavaScript method followed by the arguments of the method itself. 

In your case you have to take a look at [Map.removeLayer](https://maplibre.org/maplibre-gl-js/docs/API/classes/Map/#removelayer). As you can see the only argument ist the `layerId`, so in Python it looks like this:

```python
m = Map()
# ...
m.add_call("removeLayer", layer_id)
```

**Comment by wendellwt** [comment]
That works great!  Thank you.

BTW, I removed both the layer and the source, then added them both again.  Was that necessary?

```
def layer_update(params):

    m.add_call("removeLayer", LAYER_ID)
    m.add_call("removeSource", SOURCE_ID)

    new_url = f"https://whatever.com/{z}/{x}/{y}?{params}.tiff"

    raster_tile_source = RasterTileSource(
        tiles=[new_url],
        ...
    )
    m.add_source(SOURCE_ID, raster_tile_source)

    wx_layer = Layer(
        type=LayerType.RASTER,
        source=SOURCE_ID,
        id=LAYER_ID,
    )
    m.add_layer(wx_layer)

```
